### ISSUE TITLE
Navigation with Screen reader is hard in settings because of random loss of focus

### ISSUE THREAD

**Issue by Nardol** [home-assistant/frontend#8414]
**Checklist**

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

**Describe the issue you are experiencing**

Using Firefox or Google Chrome, when browsing the entity list, the focus moves to another place of the page, randomly. Consequence is we cannot know if we are still in the exposed entities or not.
It also happens in HA cloud main settings and in integration list.
Configuring Google Assistant by YAML make navigation possible in entities list but it make it possible in a page which becomes useless because configuring by YAML logically disable graphical settings.
I don't know if this is the same on other OS than Linux.
But I don't observe this on Android.

**Describe the behavior you expected**

When pressing down arrow, we should really move to the next line, not to some random and we should not go to the beginning of the page randomly

**Steps to reproduce the issue**

1. Using Orca, Go to the entities configuration for Google Assistant;
2. Activate browse mode;
3. Press h to go to the heading "exposed entities". With Google Chrome, we have to press Ctrl+End and use Shift+h to find the heading as the mamin menu lock the browse mode;
4. Press down arrow some times

**What version of Home Assistant Core has the issue?**

2021.2.3

**What was the last working version of Home Assistant Core?**

_No response_

**In which browser are you experiencing the issue with?**

Firefox 85.0.2 and Google Chrome

**Which operating system are you using to run this browser?**

Debian 10 (Buster)

**State of relevant entities**

```yaml
# Paste your state here.

```

**Problem-relevant frontend configuration**

```yaml
# Paste your YAML here.

```

**Javascript errors shown in your browser console/inspector**

```txt
# Paste your logs here.

```

**Comment by Nardol** [comment]
With Google Chrome it looks like better in fact. But with Firefox it is after doing no more than five down arrow presses and we are lost in the list.

**Comment by Nardol** [comment]
After more testing, it is not happening at all with Google Chrome, it is only with Firefox.

**Comment by Nardol** [comment]
Hi,

This issue still occurs.

To be more precize, this accessibility (A11y) issue occurs only on Firefox

**Comment by Nardol** [comment]
And in fact, the same issue occurs in the integrations list, which has become less accessible since we can not move between each integration since headings have been removed.

Has accessibility for screen reader been forgotten?

**Comment by Nardol** [comment]
This issue still occurs.
I have to switch from Firefox to Chrome to be able to add or remove an entity because I cannot be sure of the one I will expose or not to Google Assistant.

**Comment by Nardol** [comment]
Still the same.

**Comment by Nardol** [comment]
I'm going to ask for more information on Orca side, as this behavior also occurs on home-assistant.io.
It looks like Orca get a refresh event or something similar.

**Comment by Nardol** [comment]
I still observe this behavior.

**Comment by Nardol** [comment]
Before Home Assistant 2022.7, I also noticed this focus issue.
Now, I can read all the logbook without the cursor being moved alone.
If I am right there have been some improvment with scrolling.
If yes, could these modification be applyed also here?

**Comment by Nardol** [comment]
I have to add I also have this issue with Google Chrome now.

**Comment by Nardol** [comment]
In fact it look "worse" than I noticed, because i happens in a lot of settings pages.
I did not notice it before because I used search in page or quick navigation keys from screen reader but classic navigation is make it reproducible.

In integrations page, I move line by line after finding a title, and after one or two down arrow presses I am on the top of the integration list.

Yesterday I noticed it in devices page but since 2022.10 I cannot reproduce it anymore, table improvement with how headers might have improved the situation.

When it happens it makes navigation very hard.
@steverep do you have an idea?

For Google Assistant entities management it does not happen if I configure them by YAML so if I disable setting by UI.

In cloud settings in general navigation is hard because of this behavior.

**Comment by steverep** [comment]
I have to admit I'm very confused by the issue you are reporting.  Let me ask some questions to hopefully clarify...

> In integrations page, I move line by line after finding a title, and after one or two down arrow presses I am on the top of the integration list.
 Are you talking about the main integrations settings page, or the dialog to add a new integration?  Exact steps to get there would help.

> Yesterday I noticed it in devices page but since 2022.10 I cannot reproduce it anymore, table improvement with how headers might have improved the situation.

That shouldn't have remedied anything like what you are describing unless there is a really weird bug in Orca.  Again, which exact page with exact steps would help.

**Comment by Nardol** [comment]
> Are you talking about the main integrations settings page, or the dialog to add a new integration?  Exact steps to get there would help.
> 

1. go to [Settings→Devices and services](https://my.home-assistant.io/redirect/integrations/)
2. Go to the next heading (so h with NVDA, t with JAWS if I remember correctly)
3. Press down arrow more than once, problem is the number is random

At the moment it occurs, cursor jump after tabs so to the button to open the menu to display application authentication information.

> That shouldn't have remedied anything like what you are describing unless there is a really weird bug in Orca. Again, which exact page with exact steps would help.

I'm referring to [the devices tabs in integrations and devices](https://my.home-assistant.io/redirect/devices/), in this page I can now navigate using arrows without any issues, before cursor randomly jumped to the beginning so after all tabs.

**Comment by steverep** [comment]
> At the moment it occurs, cursor jump after tabs so to the button to open the menu to display application authentication information.

I have never experienced anything like that on Windows in Firefox or Chrome with NVDA.  I don't currently have a Linux VM setup like I normally do so I cannot try to reproduce with Orca.

To clarify though, are you saying the actual keyboard focus jumps or just the virtual browse cursor?  For example, if you use the tab key instead of down arrow, does the focus jump?


> I'm referring to [the devices tabs in integrations and devices](https://my.home-assistant.io/redirect/devices/), in this page I can now navigate using arrows without any issues, before cursor randomly jumped to the beginning so after all tabs.


Again, I've never experienced anything like this, but if you can definitively narrow down the version change that caused the improvement, we might be able to find a fix.

If you have a development environment setup, I would suggest checking out tags until you find the one that improves it, then checking out different commits that might contain the improvement.  Tedious work, but worth a shot.

**Comment by Nardol** [comment]
Things are beginning worse with each new monthly version, since the beginning of April I notice also this behavior in my main dashboard.

It is when using browse mode, when using the tab key I don't notice any issue.

**Comment by Nardol** [comment]
@steverep things are now really better.
I don't know why, but focus is really stable now.
Not perfect, but cursor is not moved to the beginning as it was before.
